lib/engine/game/g_1848/game.rb

```diff
@@ -701,36 +719,60 @@ def market_share_limit(corporation = nil)
         def can_take_loan?(entity)
           entity.corporation? &&
             entity.loans.size < maximum_loans(entity) &&
-            @loans.any?
+            @loans.any? &&
+            @take_out_loan_triggered
         end
 
-        def take_loan(entity, loan)
-          raise GameError, "Cannot take more than #{maximum_loans(entity)} loans" unless can_take_loan?(entity)
+        def take_loan(entity, loan, ebuy: false)
+          raise GameError, "Cannot take more than #{maximum_loans(entity)} loans" unless can_take_loan?(entity, ebuy)
 
           old_price = entity.share_price
           boe_old_price = @boe.share_price
           @boe.spend(loan.amount, entity)
-          loan_taken_stock_market_movement(entity, loan)
+          loan_taken_stock_market_movement(entity, loan, ebuy)
           log_share_price(entity, old_price)
           log_share_price(@boe, boe_old_price)
           entity.loans << loan
           @boe.loans << loan
           @loans.delete(loan)
         end
 
-        def loan_taken_stock_market_movement(entity, loan)
+        def loan_taken_stock_market_movement(entity, loan, ebuy = false)
           @log << "#{entity.name} takes a loan and receives #{format_currency(loan.amount)}"
           2.times { @stock_market.move_left(entity) }
+          @stock_market.move_left(entity) if ebuy
           @stock_market.move_right(boe)
         end
 
+        def perform_ebuy_loans(entity, remaining)
+          ebuy = true
+          while remaining.positive? && entity.share_price.price != 0
+            # if at max loans, company goes directly into receiverhsip
+            if @loans.empty?
+              @log << "There are no more loans available to force buy a train, #{entity.name} goes into receivership"
+              r, _c = entity.share_price.coordinates
+              @stock_market.move(entity, r, 0)
+              break
+            end
+            loan = @loans.first
+            take_loan(entity, loan, ebuy)
+            remaining -= loan.amount
+          end
+        end
+
         # routing logic
         def visited_stops(route)
           modified_guage_changes = get_modified_guage_distance(route)
           added_stops = modified_guage_changes.positive? ? Array.new(modified_guage_changes) { Engine::Part::City.new('0') } : []
           super + added_stops
         end
 
+        def check_distance(route, visits, _train = nil)
+          return super if route.train.name != '2E' || ghan_visited?(visits.first) || ghan_visited?(visits.last)
+
+          raise GameError, 'Route for 2E train must include Alice Springs'
```

Comment:
These game errors are shown in the route table. The string might be too long as is. Just something to check. I think "Must include Alice Springs" would be good enough if what you have is too long.

Reply:
they show up as a new line bellow the train, so there's room, but 2e is redundant.